libs/hdf-converters/src/conveyor-mapper.ts

```diff
@@ -0,0 +1,244 @@
+    const temp = result as Record<string, unknown>;
+    const sha: string = _.get(result, 'sha256') || '';
+    _.set(temp, 'filename', _.get(mappings, sha));
+    const descriptions = _.map(
+      _.get(temp, 'result.sections') as Record<string, unknown>[],
+      (val) =>
+        createDescription(
+          val as Record<string, unknown>,
+          _.get(temp, 'result.score') as number,
+          _.get(temp, 'response.milestones.service_started') as string,
+          _.get(temp, 'response.service_name') as string,
+          _.get(temp, 'response.milestones.service_completed') as string
+        )
+    );
+    if (descriptions.length == 0) {
+      descriptions.push({
+        status: ExecJSON.ControlResultStatus.Passed,
+        code_desc: 'NA',
+        start_time: _.get(temp, 'response.milestones.service_started') as string
+      });
+    }
+    _.set(temp, 'result.sections', descriptions);
+    newout.push(temp);
+  });
+  const groups = _.groupBy(newout, (result) => {
+    return _.get(result, 'response.service_name');
+  });
+  return groups;
+}
+
+function controlMappingConveyor(): MappedTransform<
+  ExecJSON.Control & ILookupPath,
+  ILookupPath
+> {
+  return {
+    id: {path: 'sha256'},
+    title: {path: 'filename'},
+    desc: '',
+    impact: {
+      path: 'result.score',
+      transformer: (value) => {
+        return value / 1000;
+      }
+    },
+    refs: [],
+    tags: {
+      archive_ts: {path: 'archive_ts'},
+      classification: {path:'classification'},
+      expiry_ts: {path: 'expiry_ts'},
+      size: {path:'size'},
+      type: {path:'type'},
+      nist: DEFAULT_STATIC_CODE_ANALYSIS_NIST_TAGS
+
+    },
+    source_location: {
+      ref: ''
+    },
+    results: [
+      {
+        path: 'result.sections',
+        status: {path: 'status'},
+        code_desc: {path: 'code_desc'},
+        start_time: {path: 'start_time'},
+        run_time: {path:'run_time'}
+      }
+    ]
+  };
+}
+
+export class ConveyorMapper extends BaseConverter {
+  data: Record<string, unknown>;
+  type: string;
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > =  {
+      passthrough: {path: 'api_response'},
+      platform: {
+        name: 'Heimdall Tools',
+        release: HeimdallToolsVersion
+      },
+      version: {path: 'api_server_version'},
+      statistics: {},
+      profiles: [
+        {
+          name: {path: 'api_response.results[0].response.service_name'},
+          version: {path: 'api_response.results[0].response.service_version'},
+          title: {path: 'api_response.params.description'},
+          supports: [],
+          attributes: [],
+          groups: [],
+          status: 'loaded',
+          controls: [
+            {
+              path: 'api_response.results',
+              ...controlMappingConveyor()
+            }
+          ],
+          sha256: ''
+        }
+      ]
+  }
+  constructor(
+    conveyor: Record<string, unknown>,
```

Comment:
should probably have better names here in the constructor to explain what 'conveyor' is and what the difference is to 'data'